sda-commons-starter-s3/src/main/java/org/sdase/commons/spring/boot/s3/config/S3Configuration.java

```diff
@@ -47,7 +49,48 @@ public AmazonS3 getAmazonS3Client() {
   }
 
   @Bean
+  @ConditionalOnMissingBean
   public S3BucketRepository s3BucketRepository(AmazonS3 getAmazonS3Client) {
     return new S3BucketRepository(getAmazonS3Client, bucketName);
   }
+
+  public String getAccessKeyId() {
+    return accessKeyId;
+  }
+
+  public void setAccessKeyId(String accessKeyId) {
+    this.accessKeyId = accessKeyId;
+  }
+
+  public String getSecretKey() {
+    return secretKey;
+  }
+
+  public void setSecretKey(String secretKey) {
+    this.secretKey = secretKey;
+  }
+
+  public String getEndpoint() {
+    return endpoint;
+  }
+
+  public void setEndpoint(String endpoint) {
+    this.endpoint = endpoint;
+  }
+
+  public String getRegion() {
+    return region;
+  }
+
+  public void setRegion(String region) {
+    this.region = region;
+  }
+
+  public String getBucketName() {
+    return bucketName;
+  }
+
+  public void setBucketName(String bucketName) {
+    this.bucketName = bucketName;
+  }
```

Comment:
As these properties are configured by Spring magic, I would not add setters here. Why do we need the getters? Isn't this configuration self contained to provide the S3 Beans?

I see the getters being used in tests. Imho, methods should not be public for tests. You may use package private getters and add `// for test only`.

Reply:
btw. removing the setters would increase the coverage a lot ;)